### ISSUE TITLE
Error compiling fragment shader for canvas_item / light

### ISSUE THREAD

**Issue by lekoder** [godotengine/godot#18479]
**Godot version:** 3.0.2

**OS/device including version:** Win10 / NVIDIA Geforce GTX1060, linux / NVIDIA Geforce 210

**Issue description:**
Adding `light` processor function to `shader_type` `canvas_item` causes GLES3 fragment compilation to fail with:
```
ERROR: ShaderGLES3::get_current_version: CanvasShaderGLES3: Fragment Program Compilation Failed:
0(282) : error C1008: undefined variable "shadow"
0(282) : error C1008: undefined variable "screen_uv"
```
Relevant shader code:
```
279: #if defined(USE_LIGHT_SHADER_CODE)
280: //light is written by the light shader
281:            light_compute(light,light_vec,light_height,light_color,light_uv,shadow,normal,uv,screen_uv,color);
282:
283: #else
```
Expected result: fragment shader compiles

First mentioned in #13927

**Steps to reproduce:**
In a scene with `sprite` and a `light2d`, create a shader material for a sprite and add a `light` processor function to a `canvas_item` shader. Error can be observed on the console:

**Minimal reproduction project:**
[light-shader.zip](https://github.com/godotengine/godot/files/1957382/light-shader.zip)
Open scene `Sprite.tscn` to see the error.

**Comment by lekoder** [comment]
Seems to be fixed in current master as of 8d199a9b2

**Comment by akien-mga** [comment]
That was a duplicate of #16904, fixed by #17391 in the master branch and 3.0.3.
